### PR TITLE
Fixes eslint warning when node type is ChainExpression

### LINKED ISSUE #19661: Bug: eslint-plugin-react-hooks optional chaining in deps
**Packages**
* `"react": "^16.13.1"`
* `"eslint-plugin-react-hooks": "^4.0.8"`

The issue #18819 still happens with `eslint-plugin-react-hooks@4.0.8`.

Using the example code from the other issue gives the following errors:

image

image

### PR DESCRIPTION
**Pull request by pfongkye**
## Summary

Fixes #19661 
When analyzing the property chain, add a check when the node type is `ChainExpression`.

## Test Plan

I tried with a project [example ](https://github.com/pfongkye/react-rollup-boilerplate) in React + TypeScript + Rollup + `eslint-plugin-react-hooks@4.1.0` and it worked well without the provided fix.

I tested with a CRA project [example ](https://github.com/pfongkye/my_app ) with the config ejected and a local `eslint-plugin-react-hooks` containing the given fix.

### CODE REVIEW

File: packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js [hunk 1]
```diff
@@ -279,6 +279,18 @@ const tests = {
         }
       `,
     },
+    {
+      code: normalizeIndent`
+      function MyComponent() {
```
Comment (kachkaev) [inline packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js:284]: Neighbouring cases have a slightly bigger offset (+ 2 spaces). Might be worth aligning 🙂

**Comment by gaearon** [comment]
What is `ChainExpression`? I can't seem to get it on ASTExplorer with the Babel parser — is this specific to ESLint parser?

**Comment by pfongkye** [comment]
Hi @gaearon 
I don't know if it's related but I found this PR from [eslint](https://github.com/eslint/eslint/issues/12642) involving optional chaining, particularly this [one](https://github.com/eslint/eslint/pull/13416)

**Comment by pfongkye** [comment]
After some investigations using my project [example](https://github.com/pfongkye/my_app), I found that the dependency on [eslint ](https://github.com/pfongkye/my_app/blob/master/package.json#L23)has a dependency on [acorn ](https://github.com/eslint/eslint/blob/master/package.json#L89) which is bumped to `acorn@7.40` on `npm install`. The latter includes support for optional chaining (since [v7.3.0](https://github.com/acornjs/acorn/releases/tag/7.3.0)).
I think that it causes the parser to use `ChainExpression` extension. I created a gist on [ASTExplorer](https://astexplorer.net/#/gist/4782c6a37fc403a901d0d4388ed35c81/2b638ea1360c9b4f15d861be96fc14efa0af6eac).

**Comment by gaearon** [comment]
Can you update React dependencies? Our goal is to make the test _before_ your fix fail. As I understand currently it passes either way.

**Comment by pfongkye** [comment]
> Can you update React dependencies? Our goal is to make the test _before_ your fix fail. As I understand currently it passes either way.

Yes. I'll do this and make the test fail and fix afterwards :+1:

**Comment by pfongkye** [comment]
- [ ] Need some refactoring
- [ ] Either update eslint@7.7.0 and fix failing tests (not related to the eslint rule tests) or create another issue (whichever works best)

**Comment by pfongkye** [comment]
I found out that having a dependency on `@babel/eslint-parser` and testing against this parser makes the tests fail as expected. I provided a fix.

Moreover, I wanted to update `eslint` dependency to v7.7.0 but got some tests failing in [no-production-logging-test.internal.js](https://github.com/facebook/react/blob/master/scripts/eslint-rules/__tests__/no-production-logging-test.internal.js):
image

and in [ESLintRulesOfHooks-test.js](https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js):
image

 I can try to fix them either in this PR or create another issue and provide another PR. Let me know what you think.

**Review by gaearon — APPROVED** [review]

**Comment by gaearon** [comment]
lgtm, thanks

**Pull request merged**